File: 101.c
#include <stdio.h>
#include <stdbool.h>

struct TreeNode
{
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
};

bool equal(struct TreeNode *a, struct TreeNode *b)
{
    if ( a && b )
    {
        if ( a->val == b->val && equal(a->left, b->right) && equal(a->right, b->left) )
            return true;
        else
            return false;
    }
    else if ( !a && !b )
        return true;
    else
        return false;
}

bool isSymmetric(struct TreeNode* root)
{
    if ( !root )
        return true;
    else
        return equal(root->left, root->right);
}

int main()
{
    return 0;
}

File: algorithms/python/101.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def isSymmetric(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        def isEqual(a, b):
            if a and b:
                return a.val == b.val and isEqual(a.left, b.right) and isEqual(a.right, b.left)
            elif not a and not b:
                return True
            else:
                return False
        if root:
            return isEqual(root.left, root.right)
        else:
            return True
